.circleci/config.yml

```diff
@@ -88,6 +88,13 @@ commands:
       - attach_workspace:
           # Must be absolute path or relative path from working_directory
           at: ~/
+      - run:
+          name: set up node and npm install
+          command: |
+            wget https://nodejs.org/dist/v12.13.0/node-v12.13.0-linux-x64.tar.xz
+            tar -xf node-v12.13.0-linux-x64.tar.xz
+            export PATH="`pwd`/node-v12.13.0-linux-x64/bin:${PATH}"
+            npm install
       - run:
```

Comment:
I don't see a full explanation of the plan here, so I'm not sure who we expect to run `npm install`, exactly?

From this change to circleCI, it seems like this direction implies all users, even emsdk users, will need to install npm and run `npm install`? If so that worries me.

Reply:
No, emsdk users will get this as a part of post-install.

Direct git users will need to run `npm install`, but we can prompt them to do this if they forget.   We will of course need to document this clearly and mention both on the mailing list and in the ChangeLog.  One nice thing about doing it with closure first is that it only effects our users who use closure, so it will be a good way to test the waters without effecting *all* users right away.

Reply:
Another nice thing about making this a post-install step and not trying to bundle it is that the `npm install` can do all kind of platform specific things such as building native executables if needs be.  With a bundled version we wouldn't have that option (unless we want commit bundle N different platform bundles)

Reply:
I think we should talk about this suggestion to use `npm` in a more general way, as I still don't think I see the big picture. I see @juj posted a long comment in https://github.com/emscripten-core/emsdk/pull/404#issuecomment-563912301 - maybe we should consolidate the discussion there?

As to that last point, though - why would native executables matter for closure or other JS things? Or is there some future use of native executables you envision?

Reply:
Replying to this in emsdk/404